### ISSUE TITLE
Is there an option to not analyze CSV file?

### ISSUE THREAD

**Issue by ArvindRSingh** [ExcelDataReader/ExcelDataReader#339]
I have a large file 200 MB+ and ExcelDataReader first analyses the file even when ExcelReaderConfiguration is passed, it takes nearly 40 seconds just to analyze. Is there any way to make ExcelDataReader not to analyze CSV file?

**Comment by andersnm** [comment]
Hi,

Such an option does not exist unfortunately. It might be worth looking into as a new feature, although with some drawbacks: it would break the RowCount property and probably leave some data inaccessible in case of inputs with variable number of columns. 

40 seconds sounds like a lot, some questions:

How many columns and rows does your CSV have? And what encoding is it in? Would be interesting to create and test a similar CSV here if time permits.

Does it help to set ExcelReaderConfiguration.AutodetectSeparators to a single character? By default the analyzer checks for 5 separators, so you could see some performance improvements by specifying the correct separator up front.

**Comment by ArvindRSingh** [comment]
Below is the CSV tested on, file uses Windows CR-LF with UTF-8
```
AccountNumber,PaymentMethod,PaymentAmount,IdentificationNumber,CardExpireOn
1,Credit Card,,10,12/20
2,,0,20,12/21
3,Cash,11,30,12/22
4,Some other method,22,40,12/23
5,Trust,33,50,12/24
1,Credit Card,,10,12/20
```
..... repeated above data lines till CSV contains 12670841 lines

Explicitly set seperator and encoding using below code:
```
reader = ExcelReaderFactory.CreateCsvReader(stream, new ExcelReaderConfiguration()
{
    AutodetectSeparators = new char[] { ',' },
    FallbackEncoding = Encoding.UTF8
});
```
Tested again it reads the whole file to analyze and takes 40 seconds, this is when the CSV and executable file is on an SSD, so using non-SSD might take even more time.

**Comment by andersnm** [comment]
Have benchmarked the latest versions using a 168mb/7.3mrows CSV based on the example data above on my computer:

|Version|Total|Open/analyze|
|-|-:|-:|
v3.4.1|61884 ms|22276 ms|
v3.4.0|153190 ms|94763 ms|

Latest version should more than halve CSV parsing time, but not sure if enough? I would prefer to avoid an option like this to keep things simple and consistent

**Comment by andersnm** [comment]
@gbaryiames I see you are working on this :-) I think we need a separate option for this because the analysis step does more than just autodetect separator (rowcount, fieldcount, encoding)

An option like "AnalyzeInitialCsvRows = 100" sounds like a good idea I didn't consider before. It can have a default value of 0 to preserve the current behavior, and if its >0, it will set RowCount to -1, and analyse up to the specified number of rows.

**Comment by danielgindi** [comment]
I think it's perfectly find to even `throw` inside the RowCount when it's not known.
Many streams and other classes in .NET behave this way, when some functionality cannot be provided. Think about seeking in an unseekable stream. (Which is actually a perfectly normal case for this library, when the input stream is unseekable).

You don't have to analyze the whole file to detect encodings or separators. First 1024 bytes are enough, and this is how it's done in most libraries.

**Comment by andersnm** [comment]
This was just released in 3.6.0. I agree with @danielgindi to throw in RowCount when this option is set, so that is how it works =)
